src/app/public/modules/inline-form/inline-form.component.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output
+} from '@angular/core';
+
+import {
+  SkyInlineFormCloseArgs
+} from './';
+
+@Component({
+  selector: 'sky-inline-form',
+  templateUrl: './inline-form.component.html',
+  styleUrls: ['./inline-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyInlineFormComponent {
+
+  @Output()
+  public closed = new EventEmitter<SkyInlineFormCloseArgs>();
+
+  @Input()
+  public showDeleteButton = false;
```

Comment:
It might be safer to create an input type called `mode`, that could be used internally to show/hide buttons, or setup the component logic in a certain way. The reason this is potentially dangerous is that it provides the consumer access to the visibility of a specific button, when we might want to show/hide that button for other modes (for example, what if we added a new button setup in the future where we didn't want a delete button to appear, but because we're providing an input to show the delete button deliberately, we've lost control on how those buttons should appear to the user).

Potential modes for the first release:

```
export enum SkyInlineFormMode = {
  Add = 0,
  Edit
}
```

```
<sky-inline-form mode="edit">
```

If you don't like the word "mode", maybe "type" could be used, similar to the toast component's `SkyToastType`.

Reply:
This also might be moot now that the config object is passed in. Could you take a look at the new implementation and let me know if you think it needs further defined? Thanks!

Reply:
This is done.